servers/physics_server_3d.cpp

```diff
@@ -398,11 +398,46 @@ void PhysicsShapeQueryResult3D::_bind_methods() {
 
 ///////////////////////////////////////
 
+RID PhysicsServer3D::shape_create(ShapeType p_shape) {
+	switch (p_shape) {
+		case SHAPE_PLANE:
+			return plane_shape_create();
+		case SHAPE_RAY:
+			return ray_shape_create();
+		case SHAPE_SPHERE:
+			return sphere_shape_create();
+		case SHAPE_BOX:
+			return box_shape_create();
+		case SHAPE_CAPSULE:
+			return capsule_shape_create();
+		case SHAPE_CYLINDER:
+			return cylinder_shape_create();
+		case SHAPE_CONVEX_POLYGON:
+			return convex_polygon_shape_create();
+		case SHAPE_CONCAVE_POLYGON:
+			return concave_polygon_shape_create();
+		case SHAPE_HEIGHTMAP:
+			return heightmap_shape_create();
+		case SHAPE_CUSTOM:
+			return custom_shape_create();
+		default:
+			return RID();
+	}
+}
+
 void PhysicsServer3D::_bind_methods() {
 #ifndef _3D_DISABLED
 
-	ClassDB::bind_method(D_METHOD("shape_create", "type"), &PhysicsServer3D::shape_create);
-	ClassDB::bind_method(D_METHOD("shape_set_data", "shape", "data"), &PhysicsServer3D::shape_set_data);
```

Comment:
Why removing these methods from script bindings? It looks like it's still supported by the server API.

Reply:
Shape create was replaced by the other ones, which are thread friendly, I left it because a lot of the C++ code uses it and replacing would be a hassle. The other one I removed by mistake, re-adding..

Reply:
Still, we will probably change how shape data is set because the current way is cumbersome.